adagrams/game.py

```diff
@@ -1,11 +1,203 @@
+import random
+
+import copy
+
+import collections
+from collections import Counter
+
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+# =============================== draw_letters ===============================
+
 def draw_letters():
-    pass
+    hand_of_letters = []                             
+    
+    letter_pool_copy = copy.deepcopy(LETTER_POOL)
+    
+    while len(hand_of_letters) < 10:                            
+        letter_dict_keys = list(LETTER_POOL.keys())
+        weight_values_list = list(LETTER_POOL.values())
+        random_letter = random.choices(
+            letter_dict_keys, 
+            weights = weight_values_list, 
+            k=1
+        )[0]
+                
+        if letter_pool_copy[random_letter] > 0:
+            hand_of_letters.append(random_letter)
+            
+            letter_pool_copy[random_letter] -= 1
+            
+    return hand_of_letters
+
 
+
+# =============================== uses_available_letters ===============================
+    
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    word = word.upper()                                     # 1                                                
+    
+    for letter in word:                                     # 2                                                  
+
+        letter_bank_counter = Counter(letter_bank)          # 3
+        
+        word_counter = Counter(word)                        # 4
+
+        if letter in letter_bank_counter:                   # 5                           
+            letter_bank_counter.subtract(word_counter)      # 6                      
+            
+            if letter_bank_counter[letter] < 0:             # 7
+                return False                                # 8
+        
+        if letter not in letter_bank:                       # 9                                   
+            return False                                    # 10                     
```

Comment:
This is really nice! Especially once you told what the `Counter` actually does! Another approach to this that doesn't create a new data structure is this here: 
```python3
def uses_available_letters(word, letter_bank):
    cap_word = word.upper()

    for letter in cap_word:
       if letter not in letter_bank or cap_word.count(letter) > letter_bank.count(letter): 
            return False
       
    return True
```